image/descriptor.go

```diff
@@ -32,6 +33,10 @@ type descriptor struct {
 	Size      int64  `json:"size"`
 }
 
+func (d *descriptor) getDigest() string {
+	return strings.Replace(d.Digest, ":", "-", -1)
```

Comment:
this actually doesn't need to be attached as a receiver method on `*descriptor`, but just accept a `string`.